pyEpiabm/pyEpiabm/sweep/assign_household_ages_sweep.py

```diff
@@ -0,0 +1,389 @@
+                         ["one_child_prob_youngest_child_under_five"])) or
+                     ((num_childs == 2) and
+                        (random.random()
+                            < self.age_params
+                            ["two_children_prob_youngest_under_five"]))) else \
+                    self.age_params["max_child_age"]
+                if ((people[0].age >= 0) and
+                        (people[0].age <= youngest_age) and
+                        (people[num_childs - 1].age
+                         <= self.age_params["max_child_age"])):
+                    break
+
+            # sets age difference of parents and children
+            parent_age_gap = (people[num_childs - 1].age - people[0].age
+                              - (self.age_params["max_parent_age_gap"]
+                              - self.age_params["min_parent_age_gap"]))
+            if parent_age_gap > 0:
+                parent_age_gap += \
+                    self.age_params["max_parent_age_gap"]  # pragma: no cover
+            else:
+                parent_age_gap = self.age_params["max_parent_age_gap"]
+            while True:
+                people[num_childs].set_random_age()
+                youngest_age = people[num_childs - 1].age
+                # makes sure age of first parent is appropriate
+                # for their children
+                if ((people[num_childs].age <= people[0].age + parent_age_gap)
+                    and (people[num_childs].age
+                         >= youngest_age
+                         + self.age_params["min_parent_age_gap"]) and
+                        (people[num_childs].age
+                         >= self.age_params["min_adult_age"])):
+                    break
+
+            # makes sure if both parents are in the same household their age
+            # makes sense
+            if ((n > num_childs + 1) and
+                    (random.random()
+                     > self.age_params["prop_other_parent_away"])):
+                while True:
+                    people[num_childs + 1].set_random_age()
+                    if ((people[num_childs + 1].age
+                        <= people[num_childs].age
+                        + self.age_params["max_MF_partner_age_gap"]) and
+                        (people[num_childs + 1].age
+                        >= people[num_childs].age
+                        - self.age_params["max_FM_partner_age_gap"]) and
+                        (people[num_childs + 1].age
+                        <= people[0].age + parent_age_gap) and
+                        (people[num_childs + 1].age
+                        >= youngest_age
+                        + self.age_params["min_parent_age_gap"]) and
+                            (people[num_childs + 1].age
+                             >= self.age_params["min_adult_age"])):
+                        break
+
+            # considers case of more than 2 adults in house such as parents
+            # and grandparents
```

Comment:
Is this possible? It looks above like a 4 person household will have at least 2 children, but I am probably misunderstanding!

Reply:
Yes this is possible. This method is for a household that is three people or bigger. So this bit of code deals with a 5 person household and 1 child or a 6 person household and 2 children for example. You're correct that this case will not be hit in a household of 4 but it is used for larger household sizes.

Reply:
Ahh that makes sense!